core/src/wsv.rs

```diff
@@ -323,48 +320,49 @@ impl WorldStateView {
     /// - If trigger execution fails
     /// - If timestamp conversion to `u64` fails
     #[iroha_logger::log(skip_all, fields(block_height))]
-    pub fn apply(&self, block: &VersionedCommittedBlock) -> Result<()> {
+    pub fn apply(&mut self, block: &VersionedCommittedBlock) -> Result<()> {
         let hash = block.hash();
         let block = block.as_v1();
         iroha_logger::prelude::Span::current().record("block_height", block.header.height);
         trace!("Applying block");
         let time_event = self.create_time_event(block)?;
-        self.events_buffer
-            .borrow_mut()
-            .push(Event::Time(time_event));
+        self.events_buffer.push(Event::Time(time_event));
 
         self.execute_transactions(block)?;
         debug!("All block transactions successfully executed");
 
         self.world.triggers.handle_time_event(time_event);
 
-        let res = self
-            .world
-            .triggers
-            .inspect_matched(|id, engine, action, event| -> Result<()> {
+        let mut triggers = self.world.triggers.clone();
+        core::mem::swap(&mut self.world.triggers, &mut triggers);
```

Comment:
So it looks like this: 
1. We create clone of `triggers` with empty `matched_ids`.
2. Then we swap actual `triggers` with the clone so that triggers witch trigger triggers will fill empty `matched_ids`.
3. Then we execute swapped `triggers` and fill `matched_ids` for `triggers` in the world.
4. Then we decrement count for `triggers` in the world.

Reply:
I will add comment with clarifications.

Reply:
If this operation is so unclear maybe create a separate method inside `TriggerSet` for it?